dnx-on-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,221 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "VMAdminPassword": {
+      "defaultValue": "PASSw0rd",
+      "type": "securestring",
+      "metadata": {
+        "description": "The admin password for the Ubuntu machine"
+      }
+    },
+    "VMAdminUsername": {
+      "defaultValue": "serveradmin",
+      "type": "string",
+      "metadata": {
+        "description": "The admin username for the Ubuntu machine"
+      }      
+    },
+    "VMName": {
+      "defaultValue": "dnx-on-ubuntu",
+      "type": "string",
+      "metadata": {
+        "description": "The VMName"
+      }      
+    }
+  },
+  "variables": {
+    "NIC name": "dnx-on-ubuntu-nic",
+    "NSG name": "dnx-on-ubuntu-nsg",
+    "PIP name": "dnx-on-ubuntu-pip",
+    "VNET name": "dnx-on-ubuntu-vnet",
+    "SA for diagnostics - name": "[concat(uniquestring(resourceGroup().id), 'diagsa')]",
+    "SA for VMs - name": "[concat(uniquestring(resourceGroup().id), 'vmsa')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[parameters('VMName')]",
+      "apiVersion": "2015-06-15",
+      "location": "westeurope",
```

Comment:
You probably want to use resourceGroup().location for resource locations so that the resources don't get deployed in westeurope all the time.